apps/gendsa.c

```diff
@@ -117,6 +117,12 @@ int gendsa_main(int argc, char **argv)
         goto end2;
 
     DSA_get0_pqg(dsa, &p, NULL, NULL);
+
+    if (BN_num_bits(p) > 8192) {
```

Comment:
I would prefer to see 8192 (and 16384 below) be a `#define` or something. Should this be enforced over other parts of the code? Should the limit here be fixed instead of just throwing an error message?